chore(dynamic-sampling): clarify develop documentation about biases (#14320)
getsentry/sentry-docs: develop-docs/application-architecture/dynamic-sampling/fidelity-and-biases.mdx

## Biases for Sampling

Dynamic Sampling uses biases to adjust how many events matching certain conditions are sampled. These biases are defined as a set of rules that Relay checks for each event. Learn more about rules [on the architecture page](/dynamic-sampling/architecture/).

> Sentry defines a set of biases that are available to all customers. Some of the biases defined by Sentry can be enabled or disabled in the UI under **Project Settings -> Performance**, while others are enabled by default and cannot be disabled. An example of how the UI looks is shown in the following screenshot (the content of this screenshot is subject to change):

Reviewer: ```suggestion
Sentry defines a set of biases that are available to all customers. Some of the biases defined by Sentry can be enabled or disabled in the UI under **Project Settings -> Performance**, while others are enabled by default and cannot be disabled.See this example of the UI (the content of this screenshot is subject to change):
```

Because release adoption varies over time, a system of _decaying_ rules interpolates between two sample rates within a specified time window and using a specified interpolation function (e.g. `linear`). The sample rate is gradually lowered as user adoption increases and the volume of samples grows.

![Sample Rate and Adoption](./images/sampleRateAndAdoption.png)

The latest release bias uses a decaying rule to interpolate between a starting and ending sample rate over a time window that is statically defined for each platform. The list of adoption times is defined [here](https://github.com/getsentry/sentry/blob/9b98be6b97323a78809a829e06dcbef26a16365c/src/sentry/dynamic_sampling/rules/helpers/time_to_adoptions.py#L25). For example, Android has a longer time window than JavaScript because Android apps generally take more time to be adopted by users.

### Prioritize Dev Environments

> This bias increases the sampling rate for traces from development environments, since this data is often more valuable for debugging. Sentry identifies development environments using a maintained and regularly updated list of known environment patterns. For these environments, the sample rate is set to 100%, so all traces are sampled.

Reviewer: ```suggestion
This bias increases the sampling rate for traces from development environments, since this data is often more valuable for debugging. Sentry identifies development environments using a regularly maintained and updated list of known environment patterns. For these environments, the sample rate is set to 100%, so all traces are sampled.
```
